src/Features/CSharp/Portable/Completion/KeywordRecommenders/InKeywordRecommender.cs

```diff
@@ -24,13 +24,34 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 IsValidContextInForEachClause(context) ||
                 IsValidContextInFromClause(context, cancellationToken) ||
                 IsValidContextInJoinClause(context, cancellationToken) ||
-                syntaxTree.IsParameterModifierContext(position, context.LeftToken, cancellationToken, includeOperators: true) ||
+                IsInParameterModifierContext(position, context) ||
                 syntaxTree.IsAnonymousMethodParameterModifierContext(position, context.LeftToken, cancellationToken) ||
                 syntaxTree.IsPossibleLambdaParameterModifierContext(position, context.LeftToken, cancellationToken) ||
                 context.TargetToken.IsConstructorOrMethodParameterArgumentContext() ||
                 context.TargetToken.IsTypeParameterVarianceContext();
         }
 
+        private static bool IsInParameterModifierContext(int position, CSharpSyntaxContext context)
+        {
+            if (context.SyntaxTree.IsParameterModifierContext(
+                    position, context.LeftToken, includeOperators: true, out var parameterIndex, out var previousModifier))
+            {
+                if (previousModifier == SyntaxKind.None)
+                {
+                    return true;
+                }
+
+                if (previousModifier == SyntaxKind.ThisKeyword &&
+                    parameterIndex == 0 &&
+                    context.SyntaxTree.IsPossibleExtensionMethodContext(context.LeftToken))
```

Comment:
the index/extension checks don't really seem super necessary.  if someone wrote 'this' in a parameter position, it doesn't hurt to offer 'in' afterwards.  your call.

Reply:
Since this keyword recommender itself is conservative and checks not only that we're inside the first parameter, but also that the method is static and class is static, I opted for the same strategy here.